### PR TITLE
(1030) Assess - Check Your Answers page

### PR DESCRIPTION
**Pull request by richpjames**
# Context
This PR adds a 'Check Your Answers' (CYA) page similar to the one shown at the end of an Application, however this page shows the answers given whilst the Assessor has worked through the Assessment journey.

# Changes in this PR
The PR begins with some minor refactoring before doing some work to ensure that the functions we use for Apply CYA pages can be used for Assess CYA pages too. As with #469 there is a fine line between reuse and duplication but in this instance I have opted for reuse.
As with most of the Assess pages we follow the pattern of using a Page class and a view which are both minimal as this page needs no user input.

## Screenshots of UI changes

Assess -- allows me to assess an application (4)

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/utils/reviewUtils.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-  rowFunction: (
-    task: Task,
-    document: ApprovedPremisesApplication | ApprovedPremisesAssessment,
-  ) => Array<SummaryListItem>,
+  application: Application | Assessment,
+  rowFunction: (task: Task, document: Application | Assessment) => Array<SummaryListItem>,
 ) => {
-  const nonCheckYourAnswersSections = Apply.sections.slice(0, -1)
+  const nonCheckYourAnswersSections = isAssessment(application)
+    ? Assess.sections.slice(1, -1)
```
Comment (pezholio) [inline server/utils/reviewUtils.ts:12]: Should we be using [`getSections`](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/f303e9bdd0ab6eb2c9e21940b2bfde345902cd56/server/utils/assessmentUtils.ts#L118) here, as if a user proceeds to reject an assessment after not getting a response to an information request (as per https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/503), then the sections they'll get back will be different.

File: cypress_shared/helpers/assess.ts [hunk 2]
```diff
@@ -238,6 +240,22 @@ export default class AseessHelper {
     tasklistPage.shouldShowTaskStatus('matching-information', 'Completed')
   }
 
+  private completeCheckYourAnswersPage() {
+    // When I click on the 'check-your-answers' link
+    cy.get('[data-cy-task-name="check-your-answers"]').click()
+
+    // Then I should be taken to the suitability assessment page
```
Comment (pezholio) [inline cypress_shared/helpers/assess.ts:247]: Should this be `Then I should be taken to the check your answers page`?

File: cypress_shared/pages/assess/checkYourAnswersPage.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { ApprovedPremisesAssessment as Assessment } from '../../../server/@types/shared'
+import CheckYourAnswers from '../../../server/form-pages/assess/checkYourAnswers/checkYourAnswersTask/checkYourAnswers'
+import AssessPage from './assessPage'
+
+export default class CheckYourAnswersPage extends AssessPage {
```
Comment (pezholio) [inline cypress_shared/pages/assess/checkYourAnswersPage.ts:5]: Should we be checking the content of this page has the stuff we expect?

**Review by pezholio — COMMENTED** [review]
Looking good - just a couple of things that I think we should address first. Feel free to push back on any of them (or all!)

File: server/utils/reviewUtils.ts (hunk 1 above)
Comment (richpjames) [reply server/utils/reviewUtils.ts:12]: Yeah that makes sense... should I the Matching information to [that filter](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/c5430817ddde3c8a96c014980548d35f7af1b4dd/server/utils/assessmentUtils.ts#L122) it doesn't make sense to me to show them Matching Info if the Application has been rejected

File: server/utils/assessments/getSections.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { ApprovedPremisesAssessment as Assessment } from '../../@types/shared'
+import Assess from '../../form-pages/assess'
+import informationSetAsNotReceived from './informationSetAsNotReceived'
+import AssessApplication from '../../form-pages/assess/assessApplication'
+
+export default (assessment: Assessment) => {
```
Comment (pezholio) [inline server/utils/assessments/getSections.ts:6]: If we're having these as one function per file, would it work to import and export them in an index.ts file? Or would that just cause the same dependency cycle issue?

File: cypress_shared/pages/page.ts [hunk 5]
```diff
@@ -158,4 +158,10 @@ export default abstract class Page {
       cy.readFile(downloadedFilename, 'binary', { timeout: 300 })
     })
   }
+
+  shouldShowCyaTitle(taskName: string, taskTitle: string) {
```
Comment (pezholio) [inline cypress_shared/pages/page.ts:162]: Can we call this `shouldShowCheckYourAnswersTitle`? It's a bit more verbose, I know, but the `cya` naming confused me a bit to start with

**Review by pezholio — APPROVED** [review]
LGTM - just some idle thoughts around the new utils, because it's getting a bit messy, but don't stress it if it's gonna cause the same problems

File: server/utils/assessments/getSections.ts (hunk 4 above)
Comment (richpjames) [reply server/utils/assessments/getSections.ts:6]: I haven't tried. I'll give it a go

File: cypress_shared/pages/page.ts (hunk 5 above)
Comment (richpjames) [reply cypress_shared/pages/page.ts:162]: Yeah fair enough!

**Pull request merged**